tests/functional/ctu/ctu_failure/test_ctu_failure.py

```diff
@@ -0,0 +1,178 @@
+#
+# -----------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -----------------------------------------------------------------------------
+""" CTU function test."""
+
+import json
+import os
+import shutil
+import subprocess
+import unittest
+import zipfile
+
+from libtest import env
+from libtest import project
+
+from libcodechecker.analyze import host_check
+
+NO_CTU_MESSAGE = "CTU is not supported"
+NO_DISPLAY_CTU_PROGRESS = "-analyzer-display-ctu-progress is not supported"
+
+
+class TestCtu(unittest.TestCase):
+    """ Test CTU functionality. """
+
+    def setUp(self):
+        """ Set up workspace."""
+
+        # TEST_WORKSPACE is automatically set by test package __init__.py .
+        self.test_workspace = os.environ['TEST_WORKSPACE']
+
+        test_class = self.__class__.__name__
+        print('Running ' + test_class + ' tests in ' + self.test_workspace)
+
+        # Get the CodeChecker cmd if needed for the tests.
+        self._codechecker_cmd = env.codechecker_cmd()
+        self.env = env.codechecker_env()
+        self.report_dir = os.path.join(self.test_workspace, 'reports')
+        os.makedirs(self.report_dir)
+
+        self.test_dir = project.path('ctu_failure')
+
+        # Get if clang is CTU-capable or not.
+        cmd = [self._codechecker_cmd, 'analyze', '-h']
+        output = subprocess.check_output(cmd, stderr=subprocess.STDOUT,
+                                         cwd=self.test_dir, env=self.env)
+        self.ctu_capable = '--ctu-' in output
+        print("'analyze' reported CTU-compatibility? " + str(self.ctu_capable))
+
+        self.ctu_has_analyzer_display_ctu_progress = \
+            host_check.has_analyzer_feature(self.__getClangSaPath(),
+                                            '-analyzer-display-ctu-progress')
+        print("Has -analyzer-display-ctu-progress? " + str(self.ctu_capable))
+
+        # Fix the "template" build JSONs to contain a proper directory
```

Comment:
I would recommend to copy the whole `ctu_failure` test project under the test workspace ad do the modifications there and not in the source tree. Later some other test might use this project which could cause conflicts.
Maybe less `os.chdir` is needed with the test project copy. I'm not a big fan of it, it can be hard to follow in which directory the scripts are running.

Reply:
Ok, I removed the cwd changing.